docs/blog/_posts/2019-12-20-21th-dotty-milestone-release.md

```diff
@@ -0,0 +1,378 @@
+
+This project contains tests to build and test a corpus of open sources Scala 2.x
+projects against Scala 3.
+
+To run the community-build on a local machine from the main Dotty repository, first fetch all the git
+submodules with `git submodule update --init` and run `sbt community-build/test`
+from the root of the dotty repo.
+
+### New Issues
+
+Firstly thank you for all the hard work in issue reporting! Being feature complete means that our
+issue tracker will now be more important than ever. We encourage you to stress
+the compiler and report self contained test-cases! Bug minimization is hard and
+a form of art! Help us unearth those nasty bugs! ✊
+
+Last but not least we restate the mission of Scala 3. Scala has pioneered the
+fusion of object-oriented and functional programming in a typed setting and Scala 3
+will be a big step towards realising the full potential of these ideas. Its main
+objectives are to:
+
+- become more opinionated by promoting programming idioms we found to work well,
+- simplify where possible,
+- eliminate inconsistencies and surprising behaviours,
+- build on strong foundations to ensure the design hangs together well,
+- consolidate language constructs to improve the language’s consistency, safety, ergonomics, and
+  performance.
+
+You can learn more about Dotty on our [website](https://dotty.epfl.ch).
+
+<!--more-->
+
+# What’s new in the 0.21.0-RC1 technology preview?
+
+## Dotty with explicit nulls and flow typing
+
+We add support for non-nullable reference types under the compiler option
+`-Yexplicit-nulls`. Nullability needs then to be expressed explicitly via unions
+(e.g. `String|Null`).
+
+This means the following code will no longer typecheck:
+
+```scala
+val x: String = null // error: found `Null`,  but required `String`
+```
+
+Instead, to mark a type as nullable we use a type union:
+
+```scala
+val x: String|Null = null // ok
+```
+
+This change affects two parts of the compiler. Firstly we have a new type
+hierarchy for `Null` and a _translation layer_ from Java types to Scala types,
+which balances soundness and usability.
+
+With this release we also introduce a flow-sensitive analysis that refines the
+type of an expression based on control-flow. In the example below `s` is
+`String|Null`. The `if` branch validates the value of `s` against `Null` so `s`
+can be safely considered `String` in that scope.
+
+```scala
+val s: String|Null = ???
+
+if (s != null) {
+  // s: String
+}
+else {
+  // s: String|Null
+}
+```
+
+Note, that more complex tests are also supported like:
+
+```scala
+val s: String|Null = ???
+val s2: String|Null = ???
+
+if (s != null && s2 != null) // s: String and s2: String
+```
+
+but also in a short-circuting manner:
+
+```scala
+val s: String|Null = ???
+
+if (s != null && s.length > 0) // s: String in `s.length > 0`
+```
+
+For more info refer to the docs on [Explicit Nulls](https://dotty.epfl.ch/docs/reference/other-new-features/explicit-nulls.html).
```

Comment:
bit surprised that `UncheckedNull` isn't mentioned? was that a conscious choice?